server/server.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/allegro/bigcache"
+)
+
+const (
+	apiVersion  = "v1"
+	apiBasePath = "/api/" + apiVersion + "/"
+
+	// path to cache.
+	cachePath = apiBasePath + "cache/"
+)
+
+var (
+	shards             int
+	maxEntriesInWindow int
+	lifeWindow         int64
+	hardMax            int
+	verbose            bool
+	maxEntrySize       int
+	logfile            string
+	port               int
+	cache              *bigcache.BigCache
```

Comment:
I prefer to not have a global cache. It makes testing harder

Reply:
I followed the [`database/sql` pattern](https://stackoverflow.com/questions/17376207/how-to-share-mysql-connection-between-http-goroutines) for multiple HTTP handlers as access will need to be shared across multiple handlers if it's going to be extensible for stats and other future features. How would you prefer cache access?